### ISSUE TITLE
Infinite Scroll elements only show once on some platforms

### ISSUE THREAD

**Issue by mdegans** [lucasmerlin/hello_egui#26]
On Microsoft Edge (macOS), Chrome (macOS) as well as macOS native. instances of `InfiniteScroll` only show up the first time the executable is run. To repro, my website is [here](https://mdegans.github.io/website/) and the commit to repro [here](https://github.com/mdegans/website/commit/8dee925a19a4f5293312cc82b1c1f03b968b85d6).

The first time the site is run, the infinite scroll elements display:
<img width="502" alt="Screenshot 2024-06-04 at 2 41 16 PM" src="https://github.com/lucasmerlin/hello_egui/assets/47511965/262073e2-8c10-487f-88c0-748979d7454a">

After a refresh, the area where they should be shows as blank:
<img width="500" alt="Screenshot 2024-06-04 at 2 41 38 PM" src="https://github.com/lucasmerlin/hello_egui/assets/47511965/71d5c757-69e3-471e-9d0f-539ba324babe">

Clearing the site cookies resolves the issue once, but after a refresh, the widgets are gone again. This happens on native as well (at least macOS).

To aid in debugging and to point you to where the elements are used in my source, egui settings are available under the right navbar.

<img width="500" alt="Screenshot 2024-06-04 at 2 45 04 PM" src="https://github.com/lucasmerlin/hello_egui/assets/47511965/ec3bfab9-decf-4afd-89e6-d1adcd016354">

I'll investigate this myself but would appreciate input if the cause is known or if I'm "holding it wrong".

**Comment by mdegans** [comment]
Turning off persistence solved this. Not really a fix, but for me it's fine since nothing needs to persist.

**Comment by lucasmerlin** [comment]
Hmm, I guess the size of some widgets get's persisted by egui in a weird state, I had something like this before, where the size of some widget was nan in egui's persistent memory and only resetting memory would bring it back. Not sure if this is something caused directly by infinite scroll. But I never tried using the infinite scroll with collapsible headers, I'll try and see if that breaks something.

**Comment by mdegans** [comment]
That would make sense. I just investigated the `.ron` with a commit [just before I disabled persistance](https://github.com/mdegans/website/tree/8dee925a19a4f5293312cc82b1c1f03b968b85d6). There's almost no app state, which is why I disabled it. The elements that are broken aren't serialized. By deleting keys, I found the line that was causing the problem:

```ron
{
...
    "app": "(navbar:(expanded:false,selected_tab:Info,state:{}),central:())",
...
```

No NAN/Infinity, but somehow that breaks it. If I delete that and everything defaults, it works. And now that I check, I have derived `Default`, which skips the `new` which loads the actual data. But If i delete `Panel`, this implementation of `Default` is used which loads the data correctly:

```rust
impl Default for Panel {
    fn default() -> Self {
        Self {
            commonmark_cache: Default::default(),
            gallery: Gallery::new(include_str!("../../assets/gallery.json")),
            doggos: Gallery::new(include_str!("../../assets/doggos.json")),
            blog: Blog::new(),
        }
    }
}
```

So, this is on me. I could fix this by just removing serde from the parent `Panel` and then `Default` gets called every time. Closing as the issue is not with your library.
